### PR TITLE
feat(aws-s3): Add DeployAction for codepipeline

### LINKED ISSUE #1585: Support S3 Deploy Action
CDK should support the new DeployToS3 Action. If nobody is working on it, I can send a PR within the next days.

### PR DESCRIPTION
**Pull request by hoegertn**
----

### Pull Request Checklist

* [x] Testing
  - Unit test added
* [X] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [X] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

fixes #1585 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-s3/lib/pipeline-action.ts [hunk 1]
```diff
@@ -62,3 +62,54 @@ export class PipelineSourceAction extends codepipeline.SourceAction {
...
+ * Construction properties of the {@link PipelineDeployAction S3 deploy Action}.
+ */
+export interface PipelineDeployActionProps extends codepipeline.CommonActionProps,
+  codepipeline.CommonActionConstructProps {
+  /**
+   * The Amazon S3 bucket that is the deploy target
+   */
+  bucket: IBucket;
+  /**
+   * Should the deploy action extract the artifact before deploying to Amazon S3. Defaults to true
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-s3/lib/pipeline-action.ts]: Document the default as

```
@default true
```

File: packages/@aws-cdk/aws-s3/lib/pipeline-action.ts [hunk 2]
```diff
@@ -62,3 +62,54 @@ export class PipelineSourceAction extends codepipeline.SourceAction {
...
+   */
+  extract?: boolean;
+  /**
+   * The key of the target object. This is required if extract is false.
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-s3/lib/pipeline-action.ts]: Same

**Comment by skinny85** [comment]
@rix0rrr ping

**Comment by rix0rrr** [comment]
LGTM to me too

**Pull request merged**